src/transformers/models/seamless_m4t/modeling_seamless_m4t.py

```diff
@@ -3051,8 +3051,9 @@ class SeamlessM4TForSpeechToText(SeamlessM4TPreTrainedModel):
     def __init__(self, config: SeamlessM4TConfig):
         super().__init__(config)
 
+        self.shared = nn.Embedding(config.vocab_size, config.hidden_size, config.pad_token_id)
         self.speech_encoder = SeamlessM4TSpeechEncoder(config)
-        self.text_decoder = SeamlessM4TDecoder(config)
+        self.text_decoder = SeamlessM4TDecoder(config, self.shared)
```

Comment:
If I understand it correctly, this should also appears on the `SeamlessM4TForSpeechToSpeech` class right ?

Reply:
I'm not familiar with the models, but Iḿ guessing the text vocabulary is not needed there.

Reply:
The speech encoder doesn't have an embedding layer so need for a shared embeddings with it. The text decoder still needs it though

Reply:
You were correct I didn't read properly that model I Supposed the SpeechDecoder decoded directly to voice, but it's actually being vocoded from the text, you are correct. However my tests do pass locally... Very odd.

Reply:
OMG the test passes on CUDA but not CPU..... Witchcraft